core/src/main/java/org/apache/accumulo/core/file/FileOperations.java

```diff
@@ -55,23 +56,24 @@ public static FileOperations getInstance() {
    */
 
   public abstract FileSKVIterator openReader(String file, Range range, Set<ByteSequence> columnFamilies, boolean inclusive, FileSystem fs, Configuration conf,
-      AccumuloConfiguration tableConf) throws IOException;
+      RateLimiter readLimiter, AccumuloConfiguration tableConf) throws IOException;
 
   public abstract FileSKVIterator openReader(String file, Range range, Set<ByteSequence> columnFamilies, boolean inclusive, FileSystem fs, Configuration conf,
-      AccumuloConfiguration tableConf, BlockCache dataCache, BlockCache indexCache) throws IOException;
+      RateLimiter readLimiter, AccumuloConfiguration tableConf, BlockCache dataCache, BlockCache indexCache) throws IOException;
 
   /**
    * Open a reader that fully support seeking and also enable any optimizations related to seeking, like bloom filters.
    *
    */
 
-  public abstract FileSKVIterator openReader(String file, boolean seekToBeginning, FileSystem fs, Configuration conf, AccumuloConfiguration acuconf)
-      throws IOException;
+  public abstract FileSKVIterator openReader(String file, boolean seekToBeginning, FileSystem fs, Configuration conf, RateLimiter readLimiter,
```

Comment:
I was thinking perhaps a fluent syntax for opening readers/writers would make sense.

E.g. so that it admits syntax like the following:
```java
FileSKVIterator iterator=fileOperations.openReader(accumuloConfiguration)
    .ofFile(filename, fileSystem, fsConfiguration)
    .withRateLimiter(rateLimiter) // optional
    .withCache(dataCache, indexCache) // optional
    .forScanning(range, columnFamilies, inclusive);
```

Unfortunately, introducing such a change seemed a bit much for the scope of what I wanted to accomplish.